fmp_server.py

```diff
@@ -388,37 +389,37 @@ def recvReqCircleCreate(self, packet_id, data, seq):
 
         city = self.session.get_city()
         circle, circle_index = city.get_first_empty_circle()
+        with circle.lock():
```

Comment:
The `with` statement should be moved closer to its use, just above:
> circle.leader = self.session

Reply:
Done